File: R/telegramBot.R
library(telegram.bot)

# Установите ваш токен бота
token <- 'YOUR_TELEGRAM_BOT_TOKEN'

# Создайте объект бота
bot <- Bot(token = token)

# Функция приветствия для команды /start
startHandler <- function(bot, update) {
  text <- sprintf("Hello, I'm Zero Bot. 👋")
  sendMessage(bot, chat_id = update$message$chat_id, text = text)
}

# Добавьте команду /start
add_handler(bot, command = 'start', callback = startHandler)

# Запустите бота
while(TRUE) {
  updates <- getUpdates(bot)
  if (length(updates$result) > 0) {
    handle_update(bot, updates$result)
    bot$offset <- updates$result[[length(updates$result)]]$update_id + 1
  }
  Sys.sleep(1)
}

File: TypeScript/telegram_bot.ts
import { Telegraf } from 'telegraf';

const TOKEN = "YOUR_TELEGRAM_BOT_TOKEN";
const LOG_PATH = "./logs/default_log.txt";

const bot = new Telegraf(TOKEN);

const logMessage = (ctx: any) => {
    const message = ctx.message;
    const user_info = `${new Date(message.date * 1000).toISOString()} | User ID: ${message.from.id} | Username: ${message.from.username} | First Name: ${message.from.first_name} | Last Name: ${message.from.last_name} | Language: ${message.from.language_code}`;
    const chat_info = `Chat Type: ${message.chat.type} | Chat ID: ${message.chat.id} | Chat Title: ${message.chat.title || 'N/A'}`;
    const log_text = `🔹${user_info} | Chat Info: [${chat_info}] | Message Text: ${message.text}\n`;

    require('fs').appendFileSync(LOG_PATH, log_text);
}

bot.start((ctx) => {
    ctx.reply(`Hello, I'm Zero Bot =>\n👤 User ID: ${ctx.message?.from.id}\n👥 Chat ID: ${ctx.message?.chat.id}`);
    logMessage(ctx);
});

bot.on('message', (ctx) => logMessage(ctx));

bot.launch();
